x/finality/keeper/power_dist_change.go

```diff
@@ -326,6 +345,20 @@ func (k Keeper) ProcessAllPowerDistUpdateEvents(
 		}
 	}
 
+	k.processBtcDelegations(ctx, cacheFpByBtcPkHex, newActiveBtcDels, func(fp, del sdk.AccAddress, sats uint64) {
+		err := k.IncentiveKeeper.BtcDelegationActivated(ctx, fp, del, sats)
+		if err != nil {
+			panic(err)
+		}
+	})
+
+	k.processBtcDelegations(ctx, cacheFpByBtcPkHex, newUnbondedBtcDels, func(fp, del sdk.AccAddress, sats uint64) {
+		err := k.IncentiveKeeper.BtcDelegationUnbonded(ctx, fp, del, sats)
```

Comment:
nop, it is not
If we call twice, it will reduce the voting power twice

Reply:
> Asking, as there is slight edge case in unbonding in which we can process two unboding events for one delegation.

Can you give more details about this one?

Reply:
okay, this would break the reward tracker 
checking what could I do here

Reply:
The ubonding events for the reward tracker was modified to be processed only to the cases where the btc delegation transactions are already inside `FinalityProviderDistInfo`

https://github.com/babylonlabs-io/babylon/blob/9ee3ba50f54f8f09257aef4a8e925da010c814a4/x/finality/keeper/power_dist_change.go#L295

@gitferry do you see any problem with this?

Reply:
I think this pr works because the unbonded delegation will not appear in the cache so that unbonding events will processed twice by the incentive module

Reply:
it shouldn't be processed twice, only once

Reply:
I will remove the btcDelegations inside the `FinalityProviderDistInfo` and add a new key inside the `x/finality` that stores the tx hash of already processed unbonded events so that expired with early unbending can be dropped when received by the second time

What do you guys think? @babylonlabs-io/core-dev

Reply:
sorry, typo: `so that unbonding events will processed twice by the incentive module` -> `so that unbonding events will not be processed twice by the incentive module`

I think the current impl is good

Reply:
So, maybe the removal of the BtcDelegations will be done in a further PR 
cc @SebastianElvis

Reply:
That makes total sense to me, I will make the modifications

Reply:
modified at https://github.com/babylonlabs-io/babylon/pull/306/commits/dcee28ef1438a50c6fdc060b4e559993aa3151a3